docs/unit-test-mocking.md

```diff
@@ -0,0 +1,186 @@
+# Unit Test Mocking
+
+Relationship between mockable functions, mock groups, and source files:
+- Mockable functions are organized into groups.
+- Groups are implemented in `.c` source files and declared in `.h` header files.
+- Each `.c` source file can implement multiple groups.
+- Each `.h` header file declares at most one group.
+
+Limitations:
+- Inline functions cannot be mocked.
+
+## Annotated Templates
+
+In this example:
+- Source file `foo.c` implements groups `foo` and `foo_print`
+- Header file `foo.h` decalares group `foo`
+- Header file `foo_print.h` decalares group `foo_print`
+
+By convention, the group is named after the header file that declares it.
+
+### File `foo.c`
+
+```
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
```

Comment:
I also meant probably not worth including this too :)

Reply:
I reference it in text, so I kept it.

Reply:
Maybe also just keep the SPDX ident. I would like to change this so it is all in one comment block and not 2.